presto-main/src/main/java/com/facebook/presto/sql/planner/plan/ChildReplacer.java

```diff
@@ -1,36 +0,0 @@
-/*
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-package com.facebook.presto.sql.planner.plan;
-
-import java.util.List;
-
-public class ChildReplacer
-{
-    private ChildReplacer()
-    {
-    }
-
-    /**
-     * Return an identical copy of the given node with its children replaced
-     */
-    public static PlanNode replaceChildren(PlanNode node, List<PlanNode> children)
-    {
-        for (int i = 0; i < node.getSources().size(); i++) {
-            if (children.get(i) != node.getSources().get(i)) {
-                return node.replaceChildren(children);
```

Comment:
Since `PlanNodes`  don't have proper `equal` and `hashCode`, then identity reference might be used to compare two plan nodes. Re-instantiating the same plan will make this assumption invalid. I don't know if that is a problem in Presto though.